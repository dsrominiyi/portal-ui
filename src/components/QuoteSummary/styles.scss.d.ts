export const quoteSummary: string;
export const rates: string;
