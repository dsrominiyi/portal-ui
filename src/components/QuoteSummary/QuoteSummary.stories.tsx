import React from 'react';
import { storiesOf } from '@storybook/react';

import QuoteSummary from '.';

const stories = storiesOf('QuoteSummary', module);

const render = (date: string, rate?: number) => (props?: any) => (
  <QuoteSummary key={props && props.key} date={date} rate={rate} />
);

export const storyRenders = {
  withoutRate: render('12/12/2019'),
  withRate: render('12/12/2019', 56.59)
};

stories.add('Without rate', storyRenders.withoutRate);
stories.add('With rate', storyRenders.withRate);
