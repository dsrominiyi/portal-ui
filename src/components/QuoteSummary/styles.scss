.quote-summary {
  text-align: center;
  font-size: 1rem;
  line-height: 1.4;
}

.rates {
  font-size: 1.5rem;
  margin-top: 16px;
  margin-bottom: 8px;
  line-height: 1;
}
