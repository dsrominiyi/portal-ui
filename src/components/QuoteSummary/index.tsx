import React, { FunctionComponent } from 'react';

import styles from './styles.scss';

interface Props {
  date: string;
  rate?: number;
}

export const QuoteSummary: FunctionComponent<Props> = ({ date, rate }: Props) => {
  const rateText = rate ? ` \u00A3${rate}` : 'No Rates';

  return (
    <div className={styles.quoteSummary}>
      <div>Inception Date</div>
      <div>{date}</div>
      <div className={styles.rates}>{rateText}</div>
    </div>
  );
};

QuoteSummary.displayName = 'QuoteSummary';
export default QuoteSummary;
