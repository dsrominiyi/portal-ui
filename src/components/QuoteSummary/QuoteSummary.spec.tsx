import { storyRenders } from './QuoteSummary.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('QuoteSummary', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
