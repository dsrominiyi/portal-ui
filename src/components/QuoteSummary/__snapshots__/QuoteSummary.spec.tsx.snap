// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteSummary component story renders match snapshots: withRate 1`] = `
<div
  className="quoteSummary"
>
  <div>
    Inception Date
  </div>
  <div>
    12/12/2019
  </div>
  <div
    className="rates"
  >
     Â£56.59
  </div>
</div>
`;

exports[`QuoteSummary component story renders match snapshots: withoutRate 1`] = `
<div
  className="quoteSummary"
>
  <div>
    Inception Date
  </div>
  <div>
    12/12/2019
  </div>
  <div
    className="rates"
  >
    No Rates
  </div>
</div>
`;
